apps/builder/app/shared/nano-states/nano-states.ts

```diff
@@ -44,13 +47,84 @@ export const useSetTreeId = (treeId: Tree["id"]) => {
   });
 };
 
-export const rootInstanceContainer = atom<Instance | undefined>();
-export const useRootInstance = () => useValue(rootInstanceContainer);
-export const useSetRootInstance = (root: Instance) => {
+export const instancesStore = atom<Map<InstancesItem["id"], InstancesItem>>(
+  new Map()
+);
+export const useSetInstances = (
+  instances: [InstancesItem["id"], InstancesItem][]
+) => {
   useSyncInitializeOnce(() => {
-    rootInstanceContainer.set(root);
+    instancesStore.set(new Map(instances));
   });
 };
+
+export const patchInstancesMutable = (
+  rootInstance: undefined | Instance,
+  instances: Instances
+) => {
```

Comment:
Do I understand correctly that this function mutates `instances` to make them the same as `rootInstance`?

Reply:
It's temporary to not rewrite all code at once

Reply:
I also don't understand what this function is doing, please add an extensive comment above

Reply:
and we need tests for it, because this is clearly something we will have to deal with a lot

Reply:
I will remove this function within week. No need to waste time on tests.

Reply:
oh, alright then